src/providers/portion-provider.ts

```diff
@@ -202,9 +204,14 @@ export class PortionProvider implements IPortionProvider {
     }
 
     return routeWithQuotes.map((routeWithQuote) => {
+      const tokenOut =
+        routeWithQuote.tokenPath[routeWithQuote.tokenPath.length - 1];
```

Comment:
nit: do we want to be defensive and check tokenPath is not empty? Although in theory it should never be empty if it can be here.

Reply:
I think that if the tokenPath was empty we would have more issues here than just calculating the tokenOut, right? Checking if its non empty would create a fork in here for the empty and non empty path, do you feel strongly about adding that check? I could do it if you do, it just feels like if it were empty it would be caught somewhere else